pyresample/test/test_utils.py

```diff
@@ -611,3 +611,43 @@ def test_norotation_yaml(self):
         test_area = load_area(f.name, 'regionB')
         self.assertEqual(test_area.rotation, 0)
         os.remove(f.name)
+
+
+class TestNetcdfCFAreaParser(unittest.TestCase):
+    """ Test loading an area definition from netCDF/CF files """
+
+    def test_load_cf_from_wrong_filepath(self):
+        from pyresample.utils import load_cf_area
+
+        # wrong case #1: the path does not exist
+        cf_file = os.path.join(os.path.dirname(__file__), 'test_files', 'does_not_exist.nc')
+        self.assertRaises(ValueError, load_cf_area, cf_file)
+
+        # wrong case #2: the path exists, but is not a netCDF file
+        cf_file = os.path.join(os.path.dirname(__file__), 'test_files', 'areas.yaml')
+        self.assertRaises(ValueError, load_cf_area, cf_file)
+
+    def test_load_cf_parameters_errors(self):
+        from pyresample.utils import load_cf_area
+
+        cf_file = os.path.join(os.path.dirname(__file__), 'test_files', 'cf_nh25km.nc')
+
+        # try to load from a variable= that does not exist
+        self.assertRaises(ValueError, load_cf_area, cf_file, 'doesNotExist')
+
+        # try to load from a variable= that is itself is a grid_mapping, but without y= or x=
+        self.assertRaises(ValueError, load_cf_area, cf_file, 'Lambert_Azimuthal_Grid',)
+
+        # try to load using a variable= that is a valid grid_mapping container, but use wrong x= and y=
+        self.assertRaises(ValueError, load_cf_area, cf_file, 'Lambert_Azimuthal_Grid', y='time', x='xc',)
+
+    def test_load_cf_from_filepath(self):
+        from pyresample.utils import load_cf_area
+
+        cf_file = os.path.join(os.path.dirname(__file__), 'test_files', 'cf_nh25km.nc')
```

Comment:
Instead of including a binary file in repo, could you create this file on the fly for the tests ?

Reply:
I could. I'd rather not at this stage.

Once this from_cf() feature is implemented, I'd like to work on the equivalent to_cf(). Then it will be easier to create CF files on the fly.

To reduce the disk space taken by netCDF/CF files, I enforced netCDF4 compression (deflate 9) and emptied the variables manually (set all cells to _FillValue). The files are now tiny.